### DISCUSSION TITLE
Where can I find eBPF kernel programs?

### DISCUSSION THREAD

**Discussion by liusy58** [bpftrace/bpftrace#2276] [Q&A]
I'm looking for eBPF kernel programs in `bpftrace` project. I am not familiar with bpftrace, but I know that bcc will always pass a eBPF kernel program to the `bpf` constructor. So I wonder where can I find them in `bpftrace`? To be more concise, how can I want to get the bytecode of eBPF programs for `bpftrace`?

**Comment by fbs** [comment]
Not sure what you're looking for exactly but every bpf program is loaded with the `bpf` system call. The way of attaching the program to an event can vary but we mostly (only?) use libbpf and libbcc for loading.

To see bytecode produced you can use the `--emit-elf` option and use `llvm-objdump` or you can use `bpftool prog` to see all bpf programs in the kernel.

**Reply by liusy58** [reply]
yeah I can understand what you mean, I mean that I know that `bpftrace` is built upon eBPF, I think there should be some programs loaded to kernel resides in this repo, where can I find them?

**Comment by hooklab** [comment]
I know what you are looking for, it is in the kernel/bpf directory of the linux kernel source code.
